clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -227,12 +227,42 @@ public boolean ready(Node node, long now) {
     }
 
     /**
+     * Diconnects the connection to a particular node, if there is one.
+     * Any pending ClientRequests for this connection will receive disconnections.
+     *
+     * @param nodeId The id of the node
+     */
+    @Override
+    public void disconnect(String nodeId) {
+        selector.close(nodeId);
```

Comment:
I think the current approach is better because it's simpler.  NC already needs to track responses that were triggered because of version issues, so we're just using that same queue.  We can revisit this later, of course.